proto/message_contents/public_key.proto

```diff
@@ -26,6 +27,25 @@ message UnsignedPublicKey {
     }
 }
 
+message SignedPublicKeyV2 {
+    bytes key_bytes = 1;  // embeds an UnsignedPublicKey
+    AssociationProof proof = 2; // signs key_bytes
+}
+
+message SignedPublicKeyRevocation {
+    bytes key_bytes = 1;  // embeds an UnsignedPublicKey
+    AssociationProof proof = 2; // signs key_bytes
+}
```

Comment:
I'm uncertain that Revocation messages ought to support dynamic signatures.

Allowing for a association proof could provide an DOS attack surface, as a malicious app could attempt to hide a READKey revocation message in a SIWE message for example. Is this something that should be supported?

Requiring a explicit message "REVOKE XMTP KEY 0x11111...." would provide a clear visual cue about what is occurring. This would however make it difficult to revoke multiple keys at once. Thoughts?

Reply:
1) Heres an example of the SIWE which uses resources https://docs.login.xyz/general-information/siwe-overview/eip-4361#example-message-to-be-signed

2)  `isn't this symmetric between provisioning a new key, and revoking an old key?` It could be made to be symmetric.

My best argument for asymmetric process would be; If a malicious developer phishes a key by generating an 'Underhanded' SIWE, a user can revoke that key and all messages would be flagged as potentially spam.

If a malicious developer phishes a revocation, the user has no recourse. Those messages will be flagged in all other recipients. Since revocation is permanent, having an explicit revocation text (rather than a SIWE) would make it clearer to users the impact of this action.

examples:

```
AcmeApp.org wants you to sign in with your Ethereum account:
0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2

URI: https://AcmeApp.org/login
Version: 1
Chain ID: 1
Nonce: 32891756
Issued At: 2023-02-28T01:25:24Z
Resources:
- ipfs://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq/
- xmtp://revoke-key/b8bab08eb3e9301a430a4104a39004b1a57e6f6fd0f6f460f7fc805415fa31e1c9515940612486dc45ce06148ce5c15e14f3cf83a5f8114ada66a641ca30ba510168a58a2cdbe7289c584f42
``` 

```
==========================================
XMTP REVOCATION REQUEST
==========================================
Signing this request will invalidate the associated key, and mark all messages as invalid. This cannot be undone. For more information visit: 'https://xmtp.org/KeyRevocation' for more information 

KeyDetails:

createdAt: 2023-02-21
key:b8bab08eb3e9301a430a4104a39004b1a57e6f6fd0f6f460f7fc805415fa31e1c9515940612486dc45ce06148ce5c15e14f3cf83a5f8114ada66a641ca30ba510168a58a2cdbe7289c584f42
```